src/sema/ast.rs

```diff
@@ -875,6 +875,22 @@ impl Contract {
             })
             .cloned()
     }
+
+    pub fn functions_with_duplicate_names(&self, ns: &Namespace) -> Vec<usize> {
```

Comment:
`pub` function needs a useful comment. 

This function is used by soroban emit. I think emit should be looking at the public cfg functions, not contract functions.